includes/functions.php

```diff
@@ -4189,4 +4186,161 @@ function pmpro_maybe_send_wp_new_user_notification( $user_id, $level_id = null )
+            <script type="text/javascript">
+                var pmpro_recaptcha_validated = false;
+                var pmpro_recaptcha_error_msg = "<?php esc_attr_e( 'Please check the ReCAPTCHA box to confirm you are not a bot.', 'paid-memberships-pro' ); ?>";
+
+                // Validation callback.
+                function pmpro_recaptcha_validatedCallback() {
+                    // ReCAPTCHA worked.
+                    pmpro_recaptcha_validated = true;
+
+                    // Re-enable the submit button.
+                    jQuery('.pmpro_btn-submit-checkout,.pmpro_btn-submit').removeAttr('disabled');
+
+                    // Hide processing message.
+                    jQuery('#pmpro_processing_message').css('visibility', 'hidden');
+
+                    // Hide error message.
+                    if ( jQuery('#pmpro_message').text() == pmpro_recaptcha_error_msg ) {
+                        jQuery( '#pmpro_message' ).hide();
+                        jQuery( '#pmpro_message_bottom' ).hide();
+                    }
+                }
+
+                // Expiration callback.
+                function pmpro_recaptcha_expiredCallback() {
+                    pmpro_recaptcha_validated = false;
+                }
+
+                // Check validation on submit.
+                jQuery(document).ready(function(){
+                    jQuery('#pmpro_form').submit(function(event){
+                        if( pmpro_recaptcha_validated == false ) {
+                            event.preventDefault();
+
+                            // Re-enable the submit button.
+                            jQuery('.pmpro_btn-submit-checkout,.pmpro_btn-submit').removeAttr('disabled');
+
+                            // Hide processing message.
+                            jQuery('#pmpro_processing_message').css('visibility', 'hidden');
+
+                            // error message
+                            jQuery( '#pmpro_message' ).text( pmpro_recaptcha_error_msg ).addClass( 'pmpro_error' ).removeClass( 'pmpro_alert' ).removeClass( 'pmpro_success' ).hide().fadeIn();
+                            jQuery( '#pmpro_message_bottom' ).hide().fadeIn();
+
+                            return false;
+                        } else {
+                            return true;
+                        }
+                    });
+                });
+            </script>
+            <script type="text/javascript"
+                    src="https://www.google.com/recaptcha/api.js?hl=<?php echo $lang;?>">
+            </script>
+		<?php }
+	}
+}
+
+if( ! function_exists( 'recaptcha_get_html' ) ) {
+	/**
+	 * Legacy unprefixed for templates using the old recaptcha_get_html
+	 *
+	 * @param string $pubkey
+	 *
+	 * @return void
+	 */
+	function recaptcha_get_html( $pubkey ) {
+		pmpro_recaptcha_get_html( $pubkey );
+	}
+}
```

Comment:
I agree with David here RE keeping the recaptcha stuff in that file.

The other problem we have with recaptcha is that the global vars aren't prefixed. We did prefix the function name here, but some other plugins may override the $recaptcha or even  $recaptcha_validated globals, causing issues... which is one of the reasons we try to selectively load this stuff.

We also likely need to namespace/prefix the library we load to avoid conflicts there as well.

If we do that, we can load this on every page load... or at least earlier somehow.